Src/DependencyCollector/Shared/Implementation/DependencyTableStore.cs

```diff
@@ -29,8 +31,18 @@ internal static DependencyTableStore Instance
         {
            get 
            {
-              Interlocked.CompareExchange<DependencyTableStore>(ref instance, new DependencyTableStore(), null);
-              return instance;
+               if (instance == null)
+               {
+                   lock (SyncRoot)
+                   {
+                       if (instance == null)
```

Comment:
I'm not questioning the singleton, only the doubled checked locking for initialization instead of a static constructor call with the variable declaration.